### PR TITLE
Seperate the MMU and CACHE from the LSU

### PR DESCRIPTION
**Pull request by h0lyalg0rithm**
I kept the logic the same as it was earlier.
One of the design considerations was to use the BIU to communicate with the cache and MMU.However it was later decided to keep the communication simple until we design a more robust MMU design.

Same as #77 but the branch is in this repo instead of the fork

### CODE REVIEW

**Comment by knute-sifive** [comment]
Suraj, I meant to put my changes in another branch (and for you to look at/merge), but I messed up.  I can still do that if you want -- apologies for that.

I've made some modifications to your code, specifically removing the 2 stage MMU/Cache callbacks.  Look over my commit (and the commit log) for explanation.

File: core/LSU.cpp [hunk 1]
```diff
@@ -57,20 +58,16 @@ namespace olympia
...
-
-        ldst_pipeline_.registerHandlerAtStage<sparta::SchedulingPhase::PostTick>(static_cast<uint32_t>(PipelineStage::MMU_LOOKUP),
-                                              CREATE_SPARTA_HANDLER(LSU, handleMMULookupReq2_));
+        ldst_pipeline_.registerHandlerAtStage<sparta::SchedulingPhase::Update>
+            (static_cast<uint32_t>(PipelineStage::MMU_LOOKUP),
+             CREATE_SPARTA_HANDLER(LSU, handleMMULookupReq_));
 
-        ldst_pipeline_.registerHandlerAtStage<sparta::SchedulingPhase::Update>(static_cast<uint32_t>(PipelineStage::CACHE_LOOKUP),
-                                              CREATE_SPARTA_HANDLER(LSU, handleCacheLookupReq1_));
+        ldst_pipeline_.registerHandlerAtStage<sparta::SchedulingPhase::Update>
```
Comment (knute-sifive) [inline core/LSU.cpp:65]: _Might_ be able to keep these in `Tick` phase now.

File: core/LSU.cpp [hunk 2]
```diff
@@ -290,7 +296,7 @@ namespace olympia
     }
 
     void LSU::getAckFromCache_(const MemoryAccessInfoPtr &updated_memory_access_info_ptr){
-        cache_hit_ = updated_memory_access_info_ptr->getCacheState() == MemoryAccessInfo::CacheState::HIT;
+        cache_hit_ = updated_memory_access_info_ptr->isCacheHit();
```
Comment (knute-sifive) [inline core/LSU.cpp:299]: I think we can event get rid of `cache_hit_` and `mmu_hit_`

Comment (h0lyalg0rithm) [reply core/LSU.cpp:299]: I would need it again in the following Pr #85 to decide if I need to  stall the pipe or let it continue as usual

File: core/LSU.cpp (hunk 1 above)
Comment (knute-sifive) [reply core/LSU.cpp:65]: Suraj, noticed that you changed this in the other PR.  Can you change it here too (and test)?

File: core/LSU.cpp (hunk 2 above)
Comment (knute-sifive) [reply core/LSU.cpp:299]: ok

**Review by knute-sifive — APPROVED** [review]
This looks good to me.  If Kathlene or Arup can look it over and give it a 👍🏻 that'd be good.

**Pull request merged**